components/tikv_util/src/future.rs

```diff
@@ -62,3 +65,33 @@ where
         .map(|_| ());
     (rx, driver)
 }
+
+/// Polls the provided future immediately. If the future is not ready,
+/// it will register the waker. When the event is ready, the waker will
+/// be notified, then the internal future is immediately polled in the
+/// thread calling `wake()`.
+pub fn poll_future_notify(f: impl Future<Output = ()> + Send + 'static) {
+    let f: BoxFuture<'static, ()> = Box::pin(f);
+    let spawn = Mutex::new(Some(f));
+    let waker = Arc::new(BatchCommandsWaker(spawn));
+    waker.wake();
+}
+
+// BatchCommandsWaker is used to make business pool notifiy completion queues directly.
+struct BatchCommandsWaker(Mutex<Option<BoxFuture<'static, ()>>>);
```

Comment:
Can you explain more?

Reply:
I'm actually really unclear, aiui there is a distinction between objects which are `Waker`s (i.e., instances of the Waker struct) vs an object of any type which can be used as a Waker because it implements the ArcWake trait. I thought that these are `Wake` objects, but on re-reading the docs I am not sure. So maybe just ignore this :-s